es/es-server/src/main/java/org/elasticsearch/common/xcontent/XContentHelper.java

```diff
@@ -89,24 +91,41 @@ public static XContentParser createParser(NamedXContentRegistry xContentRegistry
         return convertToMap(bytes, ordered, null);
     }
 
+    public static Map<String, Object> toMap(BytesReference bytes, XContentType xContentType) {
+        try {
```

Comment:
shall we maybe utilize autoclosable on input stream?  

```
try (InputStream inputStream = getUncompressedInputStream(bytes)) {
..
```

Reply:
Good point